### PR TITLE
Forget password

### PR DESCRIPTION
**Pull request by TheoA816**
CHANGES:
- the profile thing was getting blocked by the calendar so turned it into a mui dialog
- reset password functionality

SOURCES:
- inspired by https://blog.logrocket.com/implementing-a-secure-password-reset-in-node-js

Reset Password:
- user enters wrong password when logging in. forget password link shows up
- link takes them to reset password page, fills in email
- correct email (matches a valid user's) entered, unique link sent to email
- user opens link in email containing unique token, checks database to see if token still valid 
- if token still valid, user prompted to fill in new password and updated in database

To make it work locally:
- since we don't have an email yet, add a EMAIL_USERNAME and EMAIL_PASSWORD to the .env file (any gmail account would work). The reset password links will be sent from the emails you put in
- add also to .env a RESET_TOKEN_SECRET used to generate the unique tokens for reset password which can be any string? (if it doesn't work dm me on discord and ill share what ive put for mine)

### CODE REVIEW

**Comment by Suchpuns** [comment]
In the .env file, is EMAIL_PASSWORD the app password?

**Review by Suchpuns — APPROVED** [review]
SORRY THIS REVIEW TOOK SO LONG TO DO I WAS ON HOLIDAY ;-;

This is actually cracked, don't see any issue with it. Great job!!!!

Just a small thing, google won't let me click on the link in the email, but that's probably because the link leads to localhost. So once we deploy this issue should fixed :)

tldr; LGTM!
